modules/Calendar/Controller/CalendarMailManager.class.php

```
@@ -743,6 +812,57 @@ private function getSendMailRecipients($actionId, $objEvent, $regId = 0, $objReg
+                        INNER JOIN `'.DBPREFIX.'module_calendar_registration_form_field` AS `f`
+                          ON `v`.`field_id` = `f`.`id`
+                        INNER JOIN `'.DBPREFIX.'module_calendar_registration` AS `r`
+                          ON `v`.`reg_id` = `r`.`id`
+                        WHERE `r`.`event_id` = ' . $objEvent->getId() . '
+                        AND `f`.`type` = \'mail\'';
+
+            switch ($send_invitation_to) {
+                case self::MAIL_INVITATION_TO_INACTIVE:
+                    // exclude all guests which are already registered on any list
+                    $result = $db->Execute($query);
+                    if ($result !== false) {
+                        while (!$result->EOF) {
+                            // delete all registered guests out of the recipients
+                            unset($recipients[$result->fields['mail']]);
+                            $result->MoveNext();
+                        }
+                    }
+                    break;
+                case self::MAIL_INVITATION_TO_SIGNEDIN_FILTERED:
+                    // only get signed in
+                    $query .= ' AND `r`.`type` = 1';
+                    $signedinRecipients = array();
+                    $result = $db->Execute($query);
+                    if ($result !== false) {
+                        while (!$result->EOF) {
+                            $signedinRecipients[$result->fields['mail']] = '';
+                            $result->MoveNext();
+                        }
+                    }
+                    $recipients = array_intersect_key(
+                        $recipients,
+                        $signedinRecipients
+                    );
+                    break;
+                case self::MAIL_INVITATION_TO_REGISTERED:
+                    break;
+                default:
+                    die($send_invitation_to . ' is not a valid invitation type');
```

using `die()` as script abortion is not accepted. please throw an exception instead